### PR TITLE
Refactor repetitive class name

### PR DESCRIPTION
**Pull request by kefeimo**
***Fixing a bug or adding a new feature?*** *Delete this and describe: 1) What bug or
feature this PR is addressing, 2) Why you chose the approach you did and other options
you considered, and 3) Add any additional context that may be helpful for reviewers*

1) Fixed Issue  https://github.com/tsdat/ingest-template/issues/3 (Note: the original issue has detailed description of motivation.)
2) This approach (i.e., refactor repetitive class names) is a hot fix to the repetitive-classname confusion. (Other approach include utilizing composition instead of inheritance, but require more refactor operations, and might significantly change the current workflow--though the alternative approach is a design that follows Object Oriented Principles: favor composition over inheritance, and follows the Zen of Python: flat is better than nested.)
3) To test/view the result: use this link to inspect the template-generated project. https://github.com/kefeimo/ingest-template/pull/1/files

## Description

~Concisely describe what your ingest does, what instrument(s) it handles, and what
location(s) it runs at.~
Not applicable, since this is bug fix, not developing a new ingest.

Refactor steps
* IngestPipeline → IngestPipelineUtils (started with class in .../ingest-template/utils/pipeline.py)
image
* Pipeline → PipelineCustom (started with class in .../ingest-template/ingest/example_ingest_pnnl/pipeline/pipeline.py)
image
*  Pipeline → PipelineCustom (started with class in .../ingest-template/templates/ingest/{{ cookiecutter.ingest_slug }}/pipeline/pipeline.py)
    - Note: I was going to make the class name more explicit (e.g., called PipelineSeb), but it requires some Regex operation, so in this version I don’t want to go throw that much trouble.)
image
*  Pipeline → PipelineCustom (ad-hoc, manually refactor in .../ingest-template/templates/ingest/{{ cookiecutter.ingest_slug }}/tests/* since Pycharm IDE has trouble resolve jinja-represented (i.e., {{ }} ) namespace)

## Developer Checklist
Since this is bug fix instead of a new ingest, I modified it to a counterpart checklist instead.

~* [ ] Addressed and removed all instances of `# TODO – Developer:`~
~* [ ] Removed unused or unnecessary boilerplate code~
~* [ ] Updated ingest README.md and documentation~
~* [ ] Added regex patterns in `mapping.py`~
~* [ ] Added ingest-specific test(s)~
~* [ ] Ran all tests with no failures~
~* [ ] Set the title of this PR to `Added Ingest: <Ingest Name> (#<Related Issue Number>)` and gave it a label of `ingest`~

* [x] Validated refactor result associated with all instances of `# TODO – Developer:`
~* [ ] Removed unused or unnecessary boilerplate code~
* [x] Validated refactor result associated with ingest README.md and documentation
* [x] Validated refactor result associated with regex patterns in `mapping.py`
* [x] Validated refactor result associated with ingest-specific test(s)
* [x] Ran all tests with no failures (able to generate project using the template)
~* [ ] Set the title of this PR to `Added Ingest: <Ingest Name> (#<Related Issue Number>)` and gave it a label of `ingest`~

## Additional Context

(Add any additional context that may be helpful for reviewers.)

I created a new branch from main (instead of 0.2.13), hence the "Can't automatically merge" warning.

image

Then resolved conflicts
image

### CODE REVIEW

File: utils/pipeline.py [hunk 1]
```diff
@@ -4,11 +4,54 @@
 from typing import Union, List, Dict
 
 
-class IngestPipeline(IngestPipeline):
+class IngestPipelineUtils(IngestPipeline):
+    def run(self, filepath: Union[str, List[str]]) -> xr.Dataset:
```
Comment (maxwellevin) [inline utils/pipeline.py:8]: @kefeimo can you remove the run method here? It should be inherited from the parent class. The point of this pipeline is just to add the ability to run the plots separately.

**Review by maxwellevin — CHANGES_REQUESTED** [review]
It's close, but I'd like to have the `PipelineCustom` class renamed to `CustomPipeline`. I think the `IngestPipelineUtils` class could also use a slightly different name... how about `TemplatePipeline` or `TemplateIngestPipeline`? @kefeimo what are your thoughts?

**Comment by kefeimo** [comment]
@maxwellevin Thanks for the feedback. I can use the following class names
- CustomPipeline ->(inherit) 
- TemplatePipeline ->(inherit) 
- IngestPipeline ->(inherit) 
- Pipeline ->(inherit) abc
to indicate that they belong to the xxPipeline family

File: utils/pipeline.py (hunk 1 above)
Comment (kefeimo) [reply utils/pipeline.py:8]: Can do. Basically, the `def run` in `utils/pipeline.py` and `tsdat/pipeline/ingest_pipeline.py` are identical, and there is not much override in the template.


image

**Review by maxwellevin — APPROVED** [review]
Thanks @kefeimo, these changes look good to me

**Pull request merged**